### ISSUE TITLE
Build error: Exec failed due to IOException: xcrun failed with code 1.

### ISSUE THREAD

**Issue by adamjstewart** [tensorflow/tensorflow#57631]
### Issue Type

Build/Install

### Source

source

### Tensorflow Version

2.10.0

### Custom Code

Yes

### OS Platform and Distribution

macOS 12.5.1 (Apple M1 Pro)

### Python version

3.9.13

### Bazel version

5.1.1

### GCC/Compiler version

Apple Clang 13.1.6

### CUDA/cuDNN version

N/A

### GPU model and memory

N/A

### Current Behaviour?

I'm unable to install TF 2.10.0 on macOS 12.5.1. I see the following errors during build:
```
ERROR: /private/var/folders/j1/68dlgpr91vlgs26vty2c8xk80000gn/T/spackynvkrim2/02aadb0d9aff009e669c11c7332b5f49/external/llvm-project/llvm/BUILD.bazel:172:11: Compiling llvm/lib/Support/AArch64TargetParser.cpp failed: Exec failed due to IOException: xcrun failed with code 1.
This most likely indicates that SDK version [12.5.1] for platform [MacOSX] is unsupported for the target version of xcode.
Process exited with status 1
stdout: stderr: 2022-09-06 20:02:19.765 xcodebuild[25302:641207] Requested but did not find extension point with identifier Xcode.IDEKit.ExtensionSentinelHostApplications for extension Xcode.DebuggerFoundation.AppExtensionHosts.watchOS of plug-in com.apple.dt.IDEWatchSupportCore
2022-09-06 20:02:19.765 xcodebuild[25302:641207] Requested but did not find extension point with identifier Xcode.IDEKit.ExtensionPointIdentifierToBundleIdentifier for extension Xcode.DebuggerFoundation.AppExtensionToBundleIdentifierMap.watchOS of plug-in com.apple.dt.IDEWatchSupportCore
xcodebuild: error: SDK "macosx12.5.1" cannot be located.
2022-09-06 20:02:20.133 xcodebuild[25305:641233] Requested but did not find extension point with identifier Xcode.IDEKit.ExtensionSentinelHostApplications for extension Xcode.DebuggerFoundation.AppExtensionHosts.watchOS of plug-in com.apple.dt.IDEWatchSupportCore
2022-09-06 20:02:20.133 xcodebuild[25305:641233] Requested but did not find extension point with identifier Xcode.IDEKit.ExtensionPointIdentifierToBundleIdentifier for extension Xcode.DebuggerFoundation.AppExtensionToBundleIdentifierMap.watchOS of plug-in com.apple.dt.IDEWatchSupportCore
xcodebuild: error: SDK "macosx12.5.1" cannot be located.
xcrun: error: unable to lookup item 'Path' in SDK 'macosx12.5.1'
Target //tensorflow/tools/pip_package:build_pip_package failed to build
ERROR: /private/var/folders/j1/68dlgpr91vlgs26vty2c8xk80000gn/T/ajstewart/spack-stage/spack-stage-py-tensorflow-2.10.0-d66xhjmq75ezf46u2u6vid4sjarpijbo/spack-src/tensorflow/lite/python/BUILD:68:10 Middleman _middlemen/tensorflow_Slite_Spython_Stflite_Uconvert-runfiles failed: Exec failed due to IOException: xcrun failed with code 1.
This most likely indicates that SDK version [12.5.1] for platform [MacOSX] is unsupported for the target version of xcode.
Process exited with status 1
stdout: stderr: 2022-09-06 20:02:19.765 xcodebuild[25302:641207] Requested but did not find extension point with identifier Xcode.IDEKit.ExtensionSentinelHostApplications for extension Xcode.DebuggerFoundation.AppExtensionHosts.watchOS of plug-in com.apple.dt.IDEWatchSupportCore
2022-09-06 20:02:19.765 xcodebuild[25302:641207] Requested but did not find extension point with identifier Xcode.IDEKit.ExtensionPointIdentifierToBundleIdentifier for extension Xcode.DebuggerFoundation.AppExtensionToBundleIdentifierMap.watchOS of plug-in com.apple.dt.IDEWatchSupportCore
xcodebuild: error: SDK "macosx12.5.1" cannot be located.
2022-09-06 20:02:20.133 xcodebuild[25305:641233] Requested but did not find extension point with identifier Xcode.IDEKit.ExtensionSentinelHostApplications for extension Xcode.DebuggerFoundation.AppExtensionHosts.watchOS of plug-in com.apple.dt.IDEWatchSupportCore
2022-09-06 20:02:20.133 xcodebuild[25305:641233] Requested but did not find extension point with identifier Xcode.IDEKit.ExtensionPointIdentifierToBundleIdentifier for extension Xcode.DebuggerFoundation.AppExtensionToBundleIdentifierMap.watchOS of plug-in com.apple.dt.IDEWatchSupportCore
xcodebuild: error: SDK "macosx12.5.1" cannot be located.
xcrun: error: unable to lookup item 'Path' in SDK 'macosx12.5.1'
INFO: Elapsed time: 100.341s, Critical Path: 1.19s
INFO: 97 processes: 51 internal, 46 local.
FAILED: Build did NOT complete successfully
FAILED: Build did NOT complete successfully
```

### Standalone code to reproduce the issue

I'm [trying to update](https://github.com/spack/spack/pull/32544) the [Spack](https://spack.io) package manager with the latest TF release. Just like [all prior releases](https://github.com/tensorflow/tensorflow/issues/56799), we've unfortunately been unable to compile TF on Apple M1/M2. To reproduce the issue, you can use the branch where I'm trying to add TF 2.10:

```console
$ git clone https://github.com/adamjstewart/spack.git
$ cd spack
$ git checkout packages/py-tensorflow
$ . share/spack/setup-env.sh
$ spack install py-tensorflow
```

### Relevant log output

* [build log](https://github.com/tensorflow/tensorflow/files/9501838/spack-build-out.txt)
* [build env](https://github.com/tensorflow/tensorflow/files/9501840/spack-build-env-mods.txt)

**Comment by gadagashwini** [comment]
Hi @adamjstewart, Just to make sure that you followed instructions mentioned on official Tensorflow [site](https://www.tensorflow.org/install/source#macos).

For users of Apple M1 computers, to get native performance, you'll want to follow the instructions found [here](https://developer.apple.com/metal/tensorflow-plugin/). Conda has shown to have the smoothest install. Thank you!

**Comment by adamjstewart** [comment]
@gadagashwini as I mentioned above, I'm not trying to install TF using pip/conda, I'm trying to repackage it for the Spack package manager. A package manager that downloads and installs a different package manager isn't a very good package manager. We would like to build from source so we can build binaries optimized to specific CPU/GPU hardware on supercomputers, we don't want to rely on unoptimized binaries. But the from-source build doesn't seem to be working on Apple M1 for some reason. The same build recipe works fine on x86_64 though.

**Comment by SuryanarayanaY** [comment]
Hi @adamjstewart ,

I am sure there are build problems earlier due to X_code versions on Mac. I can say this might not be problem now as I gone through Two such issues with configurations X_code 14.3,bazel 5.3.0 on Macos-13.3 able to build from source successfully.

Please refer the tickets #57914 , #60179 where users builds succeeded and probably you can also give it a try and let us know. Thanks!

**Comment by adamjstewart** [comment]
Hi @SuryanarayanaY 

I tried building with GNU coreutils but that didn't help. I still see the same error message:
```
ERROR: /private/var/folders/jv/cgkfvslj6nq1l7cw0c8c_8gm0000gn/T/spack5_64ik8l/a80a44b2cbdf4a3623613aef29c648d0/external/com_google_absl/absl/profiling/BUILD.bazel:59:11: Compiling absl/profiling/internal/exponential_biased.cc failed: Exec failed due to IOException: xcrun failed with code 1.
This most likely indicates that SDK version [13.3.1] for platform [MacOSX] is unsupported for the target version of xcode.
Process exited with status 1
stdout: stderr: 2023-04-26 10:21:06.822 xcodebuild[94731:2285264] Writing error result bundle to /var/folders/jv/cgkfvslj6nq1l7cw0c8c_8gm0000gn/T/ResultBundle_2023-26-04_10-21-0006.xcresult
xcodebuild: error: SDK "macosx13.3.1" cannot be located.
2023-04-26 10:21:08.277 xcodebuild[94873:2285548] Writing error result bundle to /var/folders/jv/cgkfvslj6nq1l7cw0c8c_8gm0000gn/T/ResultBundle_2023-26-04_10-21-0008.xcresult
xcodebuild: error: SDK "macosx13.3.1" cannot be located.
xcrun: error: unable to lookup item 'Path' in SDK 'macosx13.3.1'
Target //tensorflow/tools/pip_package:build_pip_package failed to build
ERROR: /private/var/folders/jv/cgkfvslj6nq1l7cw0c8c_8gm0000gn/T/Adam/spack-stage/spack-stage-py-tensorflow-2.12.0-euzc5objnjakkm73bpc666vzlv6uguxl/spack-src/tensorflow/lite/python/BUILD:69:10 Middleman _middlemen/tensorflow_Slite_Spython_Stflite_Uconvert-runfiles failed: Exec failed due to IOException: xcrun failed with code 1.
This most likely indicates that SDK version [13.3.1] for platform [MacOSX] is unsupported for the target version of xcode.
Process exited with status 1
stdout: stderr: 2023-04-26 10:21:06.822 xcodebuild[94731:2285264] Writing error result bundle to /var/folders/jv/cgkfvslj6nq1l7cw0c8c_8gm0000gn/T/ResultBundle_2023-26-04_10-21-0006.xcresult
xcodebuild: error: SDK "macosx13.3.1" cannot be located.
2023-04-26 10:21:08.277 xcodebuild[94873:2285548] Writing error result bundle to /var/folders/jv/cgkfvslj6nq1l7cw0c8c_8gm0000gn/T/ResultBundle_2023-26-04_10-21-0008.xcresult
xcodebuild: error: SDK "macosx13.3.1" cannot be located.
xcrun: error: unable to lookup item 'Path' in SDK 'macosx13.3.1'
```
I'm on macOS 13.3.1, bazel 5.3.0, TF 2.12.0, and Xcode 14.3.

**Comment by adamjstewart** [comment]
I'm using the [Spack](https://spack.io) package manager, specifically this [build recipe](https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/py-tensorflow/package.py).

I'll try to reproduce this without Spack to make sure it isn't something weird in our environment or patches. Currently busy with the NeurIPS deadline, but will get back to this as soon as I get a chance.

**Comment by SuryanarayanaY** [comment]
Hi @adamjstewart ,

Could you please confirm the current status. I am attaching a [reference](https://screenshot.googleplex.com/8UUHQYHE2etPj4x) for successful build on MacM1 with Tf2.12.

Thanks!

**Comment by SuryanarayanaY** [comment]
@adamjstewart ,

Could you please confirm whether you got opportunity to test the build. Please confirm.

Thanks!

**Comment by adamjstewart** [comment]
Thanks for the reminder. Following the steps from https://www.tensorflow.org/install/source, I ran:
```bash
# Install and load Python deps using Spack
git clone https://github.com/tensorflow/tensorflow.git
cd tensorflow
# Sticking with master branch for now
./configure
# Use defaults for all options
bazel build --config=nogcp --config=nonccl //tensorflow/tools/pip_package:build_pip_package
# Build the minimal package
```
I see the following errors fairly early on:
```
ERROR: Traceback (most recent call last):
	File "/private/var/tmp/_bazel_Adam/a1795e315d59972d5fef2fc11783c641/external/build_bazel_rules_apple/apple/internal/transition_support.bzl", line 547, column 46, in <toplevel>
		_apple_platform_split_transition = transition(
Error in transition: Invalid transition input '//command_line_option:incompatible_enable_apple_toolchain_resolution'. Cannot transition on --experimental_* or --incompatible_* options
...
ERROR: /Users/Adam/tensorflow/WORKSPACE:88:14: error loading package '@com_github_grpc_grpc//src/compiler': at /private/var/tmp/_bazel_Adam/a1795e315d59972d5fef2fc11783c641/external/com_github_grpc_grpc/bazel/grpc_build_system.bzl:28:6: at /private/var/tmp/_bazel_Adam/a1795e315d59972d5fef2fc11783c641/external/build_bazel_rules_apple/apple/ios.bzl:33:5: at /private/var/tmp/_bazel_Adam/a1795e315d59972d5fef2fc11783c641/external/build_bazel_rules_apple/apple/internal/ios_rules.bzl:95:5: initialization of module 'apple/internal/transition_support.bzl' failed and referenced by '//external:grpc_python_plugin'
ERROR: Analysis of target '//tensorflow/tools/pip_package:build_pip_package' failed; build aborted: 
```

**Comment by SuryanarayanaY** [comment]
@adamjstewart ,

I tried the command and found build fails. Attached below the logs for reference. Needs to check.

```
(tf-metal) suryanarayanay-macbookpro:tensorflow suryanarayanay$ bazel build --config=nogcp --config=nonccl //tensorflow/tools/pip_package:build_pip_package

Extracting Bazel installation...
Starting local Bazel server and connecting to it...
INFO: Options provided by the client:
  Inherited 'common' options: --isatty=1 --terminal_columns=80
INFO: Reading rc options for 'build' from /Users/suryanarayanay/tensorflow/.bazelrc:
  Inherited 'common' options: --experimental_repo_remote_exec
INFO: Reading rc options for 'build' from /Users/suryanarayanay/tensorflow/.bazelrc:
  'build' options: --define framework_shared_object=true --define tsl_protobuf_header_only=true --define=use_fast_cpp_protos=true --define=allow_oversize_protos=true --spawn_strategy=standalone -c opt --announce_rc --define=grpc_no_ares=true --noincompatible_remove_legacy_whole_archive --enable_platform_specific_config --define=with_xla_support=true --config=short_logs --config=v2 --define=no_aws_support=true --define=no_hdfs_support=true --experimental_cc_shared_library --experimental_link_static_libraries_once=false --incompatible_enforce_config_setting_visibility
INFO: Reading rc options for 'build' from /Users/suryanarayanay/tensorflow/.tf_configure.bazelrc:
  'build' options: --action_env PYTHON_BIN_PATH=/opt/homebrew/opt/python@3.11/bin/python3.11 --action_env PYTHON_LIB_PATH=/opt/homebrew/opt/python@3.11/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages --python_path=/opt/homebrew/opt/python@3.11/bin/python3.11
INFO: Reading rc options for 'build' from /Users/suryanarayanay/tensorflow/.bazelrc:
  'build' options: --deleted_packages=tensorflow/compiler/mlir/tfrt,tensorflow/compiler/mlir/tfrt/benchmarks,tensorflow/compiler/mlir/tfrt/jit/python_binding,tensorflow/compiler/mlir/tfrt/jit/transforms,tensorflow/compiler/mlir/tfrt/python_tests,tensorflow/compiler/mlir/tfrt/tests,tensorflow/compiler/mlir/tfrt/tests/ir,tensorflow/compiler/mlir/tfrt/tests/analysis,tensorflow/compiler/mlir/tfrt/tests/jit,tensorflow/compiler/mlir/tfrt/tests/lhlo_to_tfrt,tensorflow/compiler/mlir/tfrt/tests/lhlo_to_jitrt,tensorflow/compiler/mlir/tfrt/tests/tf_to_corert,tensorflow/compiler/mlir/tfrt/tests/tf_to_tfrt_data,tensorflow/compiler/mlir/tfrt/tests/saved_model,tensorflow/compiler/mlir/tfrt/transforms/lhlo_gpu_to_tfrt_gpu,tensorflow/core/runtime_fallback,tensorflow/core/runtime_fallback/conversion,tensorflow/core/runtime_fallback/kernel,tensorflow/core/runtime_fallback/opdefs,tensorflow/core/runtime_fallback/runtime,tensorflow/core/runtime_fallback/util,tensorflow/core/tfrt/eager,tensorflow/core/tfrt/eager/backends/cpu,tensorflow/core/tfrt/eager/backends/gpu,tensorflow/core/tfrt/eager/core_runtime,tensorflow/core/tfrt/eager/cpp_tests/core_runtime,tensorflow/core/tfrt/gpu,tensorflow/core/tfrt/run_handler_thread_pool,tensorflow/core/tfrt/runtime,tensorflow/core/tfrt/saved_model,tensorflow/core/tfrt/graph_executor,tensorflow/core/tfrt/saved_model/tests,tensorflow/core/tfrt/tpu,tensorflow/core/tfrt/utils
INFO: Found applicable config definition build:short_logs in file /Users/suryanarayanay/tensorflow/.bazelrc: --output_filter=DONT_MATCH_ANYTHING
INFO: Found applicable config definition build:v2 in file /Users/suryanarayanay/tensorflow/.bazelrc: --define=tf_api_version=2 --action_env=TF2_BEHAVIOR=1
INFO: Found applicable config definition build:nogcp in file /Users/suryanarayanay/tensorflow/.bazelrc: --define=no_gcp_support=true
INFO: Found applicable config definition build:nonccl in file /Users/suryanarayanay/tensorflow/.bazelrc: --define=no_nccl_support=true
INFO: Found applicable config definition build:macos in file /Users/suryanarayanay/tensorflow/.bazelrc: --apple_platform_type=macos --copt=-DGRPC_BAZEL_BUILD --copt=-w --define=PREFIX=/usr --define=LIBDIR=$(PREFIX)/lib --define=INCLUDEDIR=$(PREFIX)/include --define=PROTOBUF_INCLUDE_PATH=$(PREFIX)/include --cxxopt=-std=c++17 --host_cxxopt=-std=c++17
WARNING: Download from https://storage.googleapis.com/mirror.tensorflow.org/github.com/tensorflow/runtime/archive/0aaa6e679847a4eeb407136e7b0bcef93ec652e6.tar.gz failed: class java.io.FileNotFoundException GET returned 404 Not Found
WARNING: Download from https://storage.googleapis.com/mirror.tensorflow.org/github.com/llvm/llvm-project/archive/a52054cfa29d665c43141c66c20a7b8f7a96b546.tar.gz failed: class java.io.FileNotFoundException GET returned 404 Not Found
WARNING: Download from https://storage.googleapis.com/mirror.tensorflow.org/github.com/google/benchmark/archive/f7547e29ccaed7b64ef4f7495ecfff1c9f6f3d03.tar.gz failed: class java.io.FileNotFoundException GET returned 404 Not Found
WARNING: Download from https://mirror.bazel.build/github.com/bazelbuild/rules_cc/archive/081771d4a0e9d7d3aa0eed2ef389fa4700dfb23e.tar.gz failed: class java.io.FileNotFoundException GET returned 404 Not Found
WARNING: Download from https://storage.googleapis.com/mirror.tensorflow.org/github.com/openxla/stablehlo/archive/458bdd95771e9861e6488868e315a1b0340058ba.zip failed: class java.io.FileNotFoundException GET returned 404 Not Found
WARNING: Download from https://storage.googleapis.com/mirror.tensorflow.org/github.com/google/boringssl/archive/c00d7ca810e93780bd0c8ee4eea28f4f2ea4bcdc.tar.gz failed: class java.io.FileNotFoundException GET returned 404 Not Found
WARNING: Download from https://storage.googleapis.com/mirror.tensorflow.org/github.com/google/XNNPACK/archive/7adae8e6ded8fff33d92212ca1028d2419cd34d4.zip failed: class java.io.FileNotFoundException GET returned 404 Not Found
WARNING: Download from https://storage.googleapis.com/mirror.tensorflow.org/github.com/pybind/pybind11_abseil/archive/2c4932ed6f6204f1656e245838f4f5eae69d2e29.tar.gz failed: class java.io.FileNotFoundException GET returned 404 Not Found
INFO: Analyzed target //tensorflow/tools/pip_package:build_pip_package (606 packages loaded, 43903 targets configured).
INFO: Found 1 target...
ERROR: /Users/suryanarayanay/tensorflow/tensorflow/BUILD:1128:21: declared output 'tensorflow/libtensorflow_framework.2.dylib' was not created by genrule. This is probably because the genrule actually didn't create this output, or because the output was a directory and the genrule was run remotely (note that only the contents of declared file outputs are copied from genrules run remotely)
ERROR: /Users/suryanarayanay/tensorflow/tensorflow/BUILD:1128:21: Executing genrule //tensorflow:libtensorflow_framework.2.dylib_sym [for host] failed: not all outputs were created or valid
realpath: illegal option -- -
usage: realpath [-q] [path ...]
Target //tensorflow/tools/pip_package:build_pip_package failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 1411.742s, Critical Path: 65.35s
INFO: 19865 processes: 8807 internal, 11058 local.
FAILED: Build did NOT complete successfully
```

**Comment by adamjstewart** [comment]
Something must have changed because with the help of https://github.com/tensorflow/tensorflow/issues/60179#issuecomment-1491238631 I'm finally able to build from source in 2.13.0! Thanks for your help!!

**Comment by adamjstewart** [comment]
The specific error that prompted this issue:
```
xcrun: error: unable to lookup item 'Path' in SDK 'macosx12.5.1'
```
still occurs when building inside Spack. Removing the following flag and allowing bazel to auto-detect the correct version solved my issue: `--macos_sdk_version=12.5.1`.
